packages/bitget-wallet/src/lib/bitget-wallet.ts

```diff
@@ -0,0 +1,284 @@
+
+      const block = await provider.block({ finality: "final" });
+
+      const data = {
+        accountId,
+        message,
+        blockId: block.header.hash,
+        publicKey: pubKey,
+        keyType: "0",
+      };
+      const encoded = JSON.stringify(data);
+
+      const signed = await _state.wallet.signMessage(
+        encoded,
+        accountId,
+        networkId
+      );
+      return {
+        ...data,
+        signature: signed.signature,
+        keyType: signed.keyType,
+      };
+    },
+    async signAndSendTransaction({ signerId, receiverId, actions }) {
+      logger.log("signAndSendTransaction", { signerId, receiverId, actions });
+
+      const { contract } = store.getState();
+
+      if (!_state.wallet.isSignedIn() || !contract) {
+        throw new Error("Wallet not signed in");
+      }
+
+      return _state.wallet
+        .signAndSendTransaction({
+          receiverId: receiverId || contract.contractId,
+          actions: actions,
+        })
+        .then((res) => {
+          if (res.error) {
+            throw new Error(res.error);
+          } else if (res.response && "error" in res.response) {
+            throw new Error(res.response.error.message);
+          }
+
+          // Shouldn't happen but avoids inconsistent responses.
+          if (!res.response?.length) {
+            throw new Error("Invalid response");
+          }
+
+          return res.response[0];
+        });
+    },
+    async signAndSendTransactions({ transactions }) {
+      logger.log("signAndSendTransactions", { transactions });
+
+      if (!_state.wallet.isSignedIn()) {
+        throw new Error("Wallet not signed in");
+      }
+
+      return _state.wallet
+        .requestSignTransactions({
+          transactions: transformTransactions(transactions),
+        })
+        .then((res) => {
+          if (res.error) {
+            throw new Error(res.error);
+          } else if (res.response && "error" in res.response) {
+            throw new Error(res.response.error.message);
+          }
+
+          // Shouldn't happen but avoids inconsistent responses.
+          if (!res.response?.length) {
+            throw new Error("Invalid response");
+          }
+
+          return res.response;
+        });
+    },
+  };
+};
+
+export function setupBitgetWallet({
+  iconUrl = icon,
+  deprecated = false,
+}: BitgetWalletParams = {}): WalletModuleFactory<InjectedWallet> {
+  return async () => {
+    const mobile = isMobile();
+    if (mobile) {
+      return null;
+    }
+
+    const installed = await isInstalled();
+    // Add extra wait to ensure Bitget Wallet's sign in status is read from the
+    // browser extension background env.
+    // Check for isSignedIn() in only if extension is installed.
+    if (installed) {
+      await waitFor(() => !!window.bitkeep?.near?.isSignedIn()).catch(
+        () => false
+      );
+    }
+
+    return {
+      id: "bitgetWallet",
```

Comment:
Please change the wallet id to `bitget-wallet` the id must match the package-name.